app/lib/tutorial.cjsx

```diff
@@ -111,3 +105,29 @@ module.exports = React.createClass
 
   handleNextClick: ->
     @refs.stepThrough.goNext()
+
+  handleUnmount: ->
+    now = new Date().toISOString()
+    completedThisSession[@props.tutorial.id] = now
+
+    if @props.user?
+      projectPreferences = @props.preferences
+      projectPreferences ?= apiClient.type('project_preferences').create
+        links:
+          project: project.id
+        preferences: {}
+      # Build this manually. Having an index (even as a strings) keys creates an array.
+      projectPreferences.preferences ?= {}
+      projectPreferences.preferences.tutorials_completed_at ?= {}
+      projectPreferences.update "preferences.tutorials_completed_at.#{@props.tutorial.id}": now
+      projectPreferences.save()
+
+      @logToGeordi now
```

Comment:
Yeah, for purposes of behavior study, it wouldn't be useful to know whenever a logged out user views the tutorial. We don't share IP info for privacy reasons, it can't really be linked to classifications for instance. Unless geordi tracks de-identified browser sessions?